lib/inc/pxp-agent/modules/task.hpp

```diff
@@ -80,21 +63,7 @@ class Task : public PXPAgent::Module, public PXPAgent::Util::Purgeable {
     leatherman::json_container::JsonContainer selectLibFile(std::vector<leatherman::json_container::JsonContainer> const& files,
         std::string const& file_name);
 
-    void callBlockingAction(
-        const ActionRequest& request,
-        const TaskCommand &command,
-        const std::map<std::string, std::string> &environment,
-        const std::string &input,
-        ActionResponse &response);
-
-    void callNonBlockingAction(
-        const ActionRequest& request,
-        const TaskCommand &command,
-        const std::map<std::string, std::string> &environment,
-        const std::string &input,
-        ActionResponse &response);
-
-    ActionResponse callAction(const ActionRequest& request) override;
+    Util::CommandObject buildCommandObject(const ActionRequest& request) override;
```

Comment:
My approach here was to keep task-specific logic in this `buildCommandObject` method, so that any classes extending this base can do their own work there and `callAction`, `callNonBlockingAction`, and `callAction` can be reused.